source/concurrent/SynchronizedList.ooc

```diff
@@ -0,0 +1,145 @@
+import structs/[VectorList, _List]
+import threading/Mutex
+
+SynchronizedList: class <T> extends _List<T> {
+	_mutex := Mutex new()
+	_backend : VectorList<T>
+	count ::= this _backend count
+	empty ::= this _backend empty
+	pointer ::= this _backend pointer
+	init: func ~default {
+		this init(32)
+	}
+	init: func ~heap (capacity: Int, freeContent := true) {
+		this init(VectorList<T> new(capacity, freeContent))
+	}
+	init: func (=_backend)
+	free: override func {
+		this _backend free()
+		this _mutex free()
+		super()
+	}
+	add: override func (item: T) {
+		this _mutex lock()
+		this _backend add(item)
+		this _mutex unlock()
+	}
+	append: override func (other: This<T>) {
+		this _mutex lock()
+		this _backend append(other _backend)
+		this _mutex unlock()
+	}
+	insert: override func (index: Int, item: T) {
+		this _mutex lock()
+		this _backend insert(index, item)
+		this _mutex unlock()
+	}
+	remove: override func ~last -> T {
+		this _mutex lock()
+		result := this _backend remove()
+		this _mutex unlock()
+		result
+	}
+	remove: override func ~atIndex (index: Int) -> T {
+		this _mutex lock()
+		result := this _backend remove(index)
+		this _mutex unlock()
+		result
+	}
+	removeAt: override func (index: Int) {
+		this _mutex lock()
+		this _backend removeAt(index)
+		this _mutex unlock()
+	}
+	clear: override func {
+		this _mutex lock()
+		this _backend clear()
+		this _mutex unlock()
```

Comment:
`Mutex` has a `with` method which does this. I don't know if `this _mutex with(this _backend clear)` would work, though. Maybe.